### PR TITLE
[Easy] Listing required fields on object types

### LINKED ISSUE #437: [Swagger] Make required fields explicitly required
After reading the docs about making certain object properties optional, it appears all properties are optional by default. This means we should explicitly make so we will have to make all the other fields required instead:

Example: https://swagger.io/docs/specification/data-models/data-types/#required

### PR DESCRIPTION
**Pull request by bh2smith**
Closes #437 and based/depends on #391

Because 

> By default, all object properties are optional. You can specify the required properties in the required list

We have to explicitly declare which fields are required as opposed to which are nullable.

### Test Plan
Open up the rendered version of the open-api spec doc and see that there is a red asterisk next to the fields which are now explicitly required.

Before:
Screenshot 2021-03-31 at 12 12 15

After:
Screenshot 2021-03-31 at 12 11 03

### CODE REVIEW

**Comment by vkgnosis** [comment]
Reading through the openapi documentation again, I believe for json objects `nullable` and `required` are different things. The former means a key-value can be set to `null` while the latter means the whole key value must be exist.
The way we have implemented the api everything is always required but also some fields can be nullable. So I like the change of adding required but we shouldn't remove `nullable`.

**Review by vkgnosis — APPROVED** [review]

File: orderbook/openapi.yml [hunk 1]
```diff
@@ -471,6 +502,16 @@ components:
...
+      required:
+        - blockNumber
+        - logIndex
+        - orderUid
+        - owner
+        - sellToken
+        - buyToken
+        - sellAmount
+        - sellAmountBeforeFees
+        - buyAmount
```
Comment (fleupold) [inline orderbook/openapi.yml:514]: If there is no transaction hash do we return null or do we omit to serialized the field? In the former case the field would still be required, but nullable.

**Review by fleupold — APPROVED** [review]

File: orderbook/openapi.yml (hunk 1 above)
Comment (bh2smith) [reply orderbook/openapi.yml:514]: Interesting fact. We return null. Will include this then. Are there any other fields that fit this description I am missing?

Comment (bh2smith) [inline orderbook/openapi.yml:514]:
```suggestion
        - buyAmount
        - transactionHash
```

**Pull request merged**